src/header/types/accept.rs

```diff
@@ -0,0 +1,301 @@
+    fn from_iter<TIterator>(iterator: TIterator) -> Self
+    where
+        TIterator: IntoIterator<Item = MediaType>,
+    {
+        Accept(LinkedHashSet::from_iter(iterator))
+    }
+}
+
+impl TypedHeader for Accept {
+    type DecodeError = AcceptError;
+
+    /// The Accept request-header field can be used to specify certain
+    /// presentation description and parameter media types [RFC6838] that are
+    /// acceptable for the response to the DESCRIBE request.
+    /// 
+    /// ```text
+    /// Accept            =  "Accept" HCOLON
+    ///                        [ accept-range *(COMMA accept-range) ]
+    ///    accept-range      =  media-type-range [SEMI accept-params]
+    ///    accept-params     =  "q" EQUAL qvalue *(SEMI generic-param )
+    ///    qvalue            =  ( "0" [ "." *3DIGIT ] )
+    ///                      /  ( "1" [ "." *3("0") ] )
+    ///    generic-param   =  token [ EQUAL gen-value ] //no example usage so not included
+    ///    gen-value       =  token / host / quoted-string
+    ///    host              = < As defined in RFC 3986>
+    ///
+    ///    media-type-range  =  ( "*//*"
+    ///                            / ( m-type SLASH "*" )
+    ///                            / ( m-type SLASH m-subtype )
+    ///                                ) *( SEMI m-parameter )
+    ///    m-type             =  discrete-type / composite-type
+    ///    discrete-type      =  "text" / "image" / "audio" / "video"
+    ///                      /  "application" / extension-token
+    ///    composite-type   =  "message" / "multipart" / extension-token
+    ///    extension-token  =  ietf-token / x-token
+    ///    ietf-token       =  token
+    ///    x-token          =  "x-" token
+    ///    m-subtype        =  extension-token / iana-token
+    ///    iana-token       =  token
+    ///    m-parameter      =  m-attribute EQUAL m-value
+    ///    m-attribute      =  token
+    ///    m-value          =  token / quoted-string
+    ///    token           =  1*(%x21 / %x23-27 / %x2A-2B / %x2D-2E / %x30-39
+    ///                        /  %x41-5A / %x5E-7A / %x7C / %x7E) //any CHAR except CTLs or tspecials
+    /// ```
+    /// 
+    /// All values seperated with commas will be converted into [`MediaType`] type.
+    /// 
+    /// # Examples
+    /// 
+    /// ```
+    /// use std::convert::TryFrom;
+    ///
+    /// use rtsp::header::map::TypedHeader;
+    /// use rtsp::header::types::accept_ranges::RangeFormat;
+    /// use rtsp::header::types::Accept;
+    /// use rtsp::header::types::accept::*;
+    /// use rtsp::header::value::HeaderValue;
+    /// use mime::*;
+    /// use std::str::FromStr;
+    /// let raw_header: Vec<HeaderValue> = vec![];
+    /// assert_eq!(Accept::decode(&mut raw_header.iter()).unwrap(), None);
+    ///
+    /// let typed_header = vec![MediaType::new(mime::STAR_STAR, None), MediaType::new(Mime::from_str("video/mp4").unwrap(), Some(QualityParam::new(0.5)))]
```

Comment:
Can you break this line into multiple lines so it doesn't go over 100 characters.

E.g.

```
let typed_header = vec![
    MediaType::new(...),
    MediaType::new(...),
];
```